### PR TITLE
ORC-806: Upgrade to Apache POM 23

### PR DESCRIPTION
**Pull request by belugabehr**
### What changes were proposed in this pull request?
Upgrade Apache Parent POM to version 23 and clean up superfluous Maven POM elements

### Why are the changes needed?
Get benefits from upgraded maven plugins. Simplification.

### How was this patch tested?
Existing unit tests.

### CODE REVIEW

File: java/pom.xml [hunk 1]
```diff
@@ -17,7 +17,7 @@
   <parent>
     <groupId>org.apache</groupId>
     <artifactId>apache</artifactId>
-    <version>18</version>
+    <version>23</version>
```
Comment (dongjoon-hyun) [inline java/pom.xml:20]: Oh, yes. I did for this at ORC-796, but actually missed this.
- https://github.com/apache/orc/commit/f0f2e1847348333a44aa69e1c6e878b4d76e0d8e#diff-d8225ebcfc11b480a6e4f54e183b67c3ead51635a167c106d928c2abf1f9ef66

**Review by dongjoon-hyun — CHANGES_REQUESTED** [review]
BTW, CI seems to fail. Could you make CI happy?
```
Error:  Failed to execute goal org.apache.maven.plugins:maven-shade-plugin:3.1.1:shade (default) on project orc-benchmarks-spark: Error creating shaded jar: Problem shading JAR /home/runner/.m2/repository/org/glassfish/jersey/core/jersey-common/2.30/jersey-common-2.30.jar entry META-INF/versions/11/org/glassfish/jersey/internal/jsr166/SubmissionPublisher$1.class: java.lang.IllegalArgumentException -> [Help 1]
```

**Comment by dongjoon-hyun** [comment]
Gentle ping, @belugabehr .

**Comment by dongjoon-hyun** [comment]
I resolved the conflicts for you, @belugabehr .

**Review by dongjoon-hyun — COMMENTED** [review]
Could you check the build failure, @belugabehr ?

```
Error:  Failed to execute goal org.apache.maven.plugins:maven-shade-plugin:3.1.1:shade (default) on project orc-benchmarks-spark: Error creating shaded jar: Problem shading JAR /home/runner/.m2/repository/org/glassfish/jersey/core/jersey-common/2.30/jersey-common-2.30.jar entry META-INF/versions/11/org/glassfish/jersey/internal/jsr166/SubmissionPublisher$1.class: java.lang.IllegalArgumentException -> [Help 1]
```

**Comment by belugabehr** [comment]
@dongjoon-hyun Hey. I'm back.  I just merged and built locally.  We shall see.

**Comment by belugabehr** [comment]
@dongjoon-hyun Should be good to go now.  Thanks.

**Comment by dongjoon-hyun** [comment]
Welcome back! Thank you so much for updating, @belugabehr .

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM. Merged to main for Apache ORC 1.8.

**Comment by dongjoon-hyun** [comment]
Thank you for your contribution again, @belugabehr . After I check this commit's status on `main` branch, I'll backport it to branch-1.7 for Apache ORC 1.7.0. In addition, we are currently releasing Apache ORC 1.6.9. You can participate the vote.
- https://lists.apache.org/thread.html/rab78879c2ff44b0357d49b592974545d5e649852c53a696510e19bdd%40%3Cdev.orc.apache.org%3E

File: java/tools/pom.xml [hunk 2]
```diff
@@ -58,13 +58,11 @@
       <groupId>org.apache.hadoop</groupId>
       <artifactId>hadoop-common</artifactId>
       <version>${tools.hadoop.version}</version>
-      <scope>compile</scope>
```
Comment (williamhyun) [inline java/tools/pom.xml:61]: Hi, @belugabehr and @dongjoon-hyun .
This seems to break ORC tools uber jar again. I'll make a PR to recover it.

File: java/tools/pom.xml [hunk 3]
```diff
@@ -82,7 +80,6 @@
     <dependency>
       <groupId>com.google.guava</groupId>
       <artifactId>guava</artifactId>
-      <scope>compile</scope>
```
Comment (williamhyun) [inline java/tools/pom.xml:85]: This one is also required.

**Comment by dongjoon-hyun** [comment]
Oh. My bad. Thank you, @williamhyun .

**Comment by dongjoon-hyun** [comment]
Hi, All.
I'll backport this and https://github.com/apache/orc/pull/753 to `branch-1.7` for Apache ORC 1.7.0.

File: java/examples/pom.xml [hunk 4]
```diff
@@ -50,14 +50,10 @@
     <dependency>
       <groupId>org.apache.hadoop</groupId>
       <artifactId>hadoop-common</artifactId>
-      <version>${hadoop.version}</version>
-      <scope>compile</scope>
```
Comment (dongjoon-hyun) [inline java/examples/pom.xml:54]: This causes another regression. https://github.com/apache/orc/pull/808 aims to fix this.

**Pull request merged**